Bottom: 22e3b80e1c1bc2fd08f8b277615eb4de224b84af
Top:    1fc368bdbf37f2d8144d7bedf1a5c90b219688d6
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2016-03-02 01:14:51 -0300

introduce callback to get type char

---

diff --git a/include/gcc-c-fe.def b/include/gcc-c-fe.def
index 25ace1c..3f3679b 100644
--- a/include/gcc-c-fe.def
+++ b/include/gcc-c-fe.def
@@ -1,6 +1,6 @@
 /* Interface between GCC C FE and GDB  -*- c -*-
 
-   Copyright (C) 2014-2015 Free Software Foundation, Inc.
+   Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
    This file is part of GCC.
 
@@ -131,6 +131,11 @@ GCC_METHOD2 (gcc_type, int_type,
 	     int /* bool */,               /* Argument IS_UNSIGNED.  */
 	     unsigned long)                /* Argument SIZE_IN_BYTES.  */
 
+/* Return the 'char' type, a distinct type from both 'signed char' and
+   'unsigned char' returned by int_type.  */
+
+GCC_METHOD0 (gcc_type, char_type)
+
 /* Return a floating point type with the given properties.  */
 
 GCC_METHOD1 (gcc_type, float_type,
diff --git a/include/gcc-c-interface.h b/include/gcc-c-interface.h
index bb1e4e6..bee7f4e 100644
--- a/include/gcc-c-interface.h
+++ b/include/gcc-c-interface.h
@@ -1,6 +1,6 @@
 /* Interface between GCC C FE and GDB
 
-   Copyright (C) 2014-2015 Free Software Foundation, Inc.
+   Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
    This file is part of GCC.
 
@@ -41,7 +41,7 @@ struct gcc_c_context;
 
 enum gcc_c_api_version
 {
-  GCC_C_FE_VERSION_0 = 0
+  GCC_C_FE_VERSION_0 = 1
 };
 
 /* Qualifiers.  */
diff --git a/include/gcc-cp-fe.def b/include/gcc-cp-fe.def
index 4a6fb7f..7774248 100644
--- a/include/gcc-cp-fe.def
+++ b/include/gcc-cp-fe.def
@@ -1,6 +1,6 @@
 /* Interface between GCC C++ FE and GDB  -*- c -*-
 
-   Copyright (C) 2014-2015 Free Software Foundation, Inc.
+   Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
    This file is part of GCC.
 
@@ -327,6 +327,11 @@ GCC_METHOD2 (gcc_type, int_type,
 	     int /* bool */,		   /* Argument IS_UNSIGNED.  */
 	     unsigned long)                /* Argument SIZE_IN_BYTES.  */
 
+/* Return the 'char' type, a distinct type from both 'signed char' and
+   'unsigned char' returned by int_type.  */
+
+GCC_METHOD0 (gcc_type, char_type)
+
 /* Return a floating point type with the given properties.  */
 
 GCC_METHOD1 (gcc_type, float_type,
diff --git a/include/gcc-cp-interface.h b/include/gcc-cp-interface.h
index 77ad7db..8a4a849 100644
--- a/include/gcc-cp-interface.h
+++ b/include/gcc-cp-interface.h
@@ -1,6 +1,6 @@
 /* Interface between GCC C++ FE and GDB
 
-   Copyright (C) 2014-2015 Free Software Foundation, Inc.
+   Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
    This file is part of GCC.
 
@@ -41,7 +41,7 @@ struct gcc_cp_context;
 
 enum gcc_cp_api_version
 {
-  GCC_CP_FE_VERSION_0 = 0xffffffff-3
+  GCC_CP_FE_VERSION_0 = 0xffffffff-4
 };
 
 /* Qualifiers.  */
diff --git a/libcc1/libcc1plugin.cc b/libcc1/libcc1plugin.cc
index 6f7b527..b8dc70f 100644
--- a/libcc1/libcc1plugin.cc
+++ b/libcc1/libcc1plugin.cc
@@ -1,7 +1,9 @@
 /* Library interface to C front end
-   Copyright (C) 2014 Free Software Foundation, Inc.
+   Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
-   This file is part of GCC.
+   This file is part of GCC.  As it interacts with GDB through libcc1,
+   they all become single combined work, and the GNU GPL applies to it
+   as a whole.
 
    GCC is free software; you can redistribute it and/or modify it under
    the terms of the GNU General Public License as published by the Free
@@ -711,6 +713,12 @@ plugin_int_type (cc1_plugin::connection *self,
 }
 
 gcc_type
+plugin_char_type (cc1_plugin::connection *self)
+{
+  return convert_out (char_type_node);
+}
+
+gcc_type
 plugin_float_type (cc1_plugin::connection *,
 		   unsigned long size_in_bytes)
 {
diff --git a/libcc1/libcp1plugin.cc b/libcc1/libcp1plugin.cc
index 7032ac0..38542a8 100644
--- a/libcc1/libcp1plugin.cc
+++ b/libcc1/libcp1plugin.cc
@@ -2,7 +2,8 @@
    Copyright (C) 2014-2016 Free Software Foundation, Inc.
 
    This file is part of GCC.  As it interacts with GDB through libcc1,
-   they all become a single program as regards the GNU GPL's requirements.
+   they all become single combined work, and the GNU GPL applies to it
+   as a whole.
 
    GCC is free software; you can redistribute it and/or modify it under
    the terms of the GNU General Public License as published by the Free
@@ -1263,6 +1264,12 @@ plugin_int_type (cc1_plugin::connection *self,
 }
 
 gcc_type
+plugin_char_type (cc1_plugin::connection *self)
+{
+  return convert_out (char_type_node);
+}
+
+gcc_type
 plugin_float_type (cc1_plugin::connection *,
 		   unsigned long size_in_bytes)
 {
